astropy/modeling/core.py

```diff
@@ -2892,6 +2900,14 @@ def _parameter_units_for_data_units(self, input_units, output_units):
                 units_for_data[param] = units_for_data_sub[param_sub]
         return units_for_data
 
+    def deepcopy(self):
+        """
+        Return a deep copy of a compound model.
+        """
+        new_model = self.copy()
+        new_model._submodels = [model.copy() for model in self._submodels]
```

Comment:
Why not use the new `deepcopy()` method here too for copying over submodels? Isn't that more fool-proof since a submodel can also be a compound (i.e., nested compound)?